src/utils.ts

```diff
@@ -17,12 +17,7 @@ interface AddressesByNetwork {
 	alfajores?: Address,
 }
 
-export const selectAddress = async (web3: Web3, addresses: AddressesByNetwork) => {
```

Comment:
`selectAddress` is sync now that we require the chainId to be passed in directly
caller is responsible to make the minimal number of `getChainId()` calls